hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/webapp/RMWebServices.java

```diff
@@ -734,7 +739,80 @@ public Response updateAppState(AppState targetState,
 
     return Response.status(Status.OK).entity(ret).build();
   }
-  
+
+  // Copy-paste job from updateAppQueue() above . Almost exactly the same pattern as YARN-1702.
```

Comment:
I don't intend to contribute this upstream. To get this into apache, they'll likely ask for a much bigger effort to get a lot more RMAdmin Commands exposed via REST. This way, I'm isolating my code from modifications that may happen upstream, so I see value in replicating some code.